### PR TITLE
Update "Management API" and "Organizations" to explain the concepts for the CAPI generation

### PR DESCRIPTION
**Pull request by AndiDog**
### What does this PR do?

Focus on CAPI concepts and explain where vintage differs. Also quite some rewording. I only touched "Organizations" (already renamed to "Multi-tenancy" as planned) a little to not make the PR too large.

### Any background context you can provide?

https://github.com/giantswarm/roadmap/issues/2393

### Have you maintained the front matter?

Yes

### CODE REVIEW

File: src/content/platform-overview/multi-tenancy/index.md [hunk 1]
```diff
@@ -17,15 +18,15 @@ owner:
 
 <div class="well disclaimer">
 
-<i class="fa fa-warning"></i> This article covers organizations as defined in the [Management API]({{< relref "/platform-overview/management-api" >}}). These are replacing the organizations as used with the [REST API]({{< relref "/use-the-api/rest-api/index.md">}}). Please make sure you read the details about this transition carefully, especially the [roadmap](#roadmap) section, to understand the ramifications for you and your end users.
+<i class="fa fa-warning"></i> This article covers organizations as defined in the [Management API]({{< relref "/platform-overview/management-api" >}}). These are replacing the organizations as used with the [REST API]({{< relref "/use-the-api/rest-api/index.md">}}). Existing customers of the REST API should be aware of this transition.
```
Comment (stone-z) [inline src/content/platform-overview/multi-tenancy/index.md:21]:
```suggestion
<i class="fa fa-warning"></i> This article covers organizations as defined in the [Management API]({{< relref "/platform-overview/management-api" >}}). These are replacing the organizations formerly used with the [REST API]({{< relref "/use-the-api/rest-api/index.md">}}). Existing users of the REST API should be aware of this transition.
```

File: src/content/platform-overview/multi-tenancy/index.md [hunk 2]
```diff
@@ -88,13 +89,20 @@ Organizations are transitioning from being managed completely by microservices b
 
 If the concept of custom resources (CR) and custom resource definitions (CRD) is new to you: Kubernetes allows to define [arbitrary objects](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to be handled via the Kubernetes API. The schema of such an object is specified by a custom resource definition. The actual objects are called the custom resources.
 
-Giant Swarm management clusters provide a CRD named `Organization` (long form: `organizations.security.giantswarm.io`). An organization is defined simply by a custom resource using that CRD, which we'll call an "organization CR" here for brevity.
+Giant Swarm management management clusters provide a CRD named `Organization` (long form: `organizations.security.giantswarm.io`, [schema documentation]({{< relref "/use-the-api/management-api/crd/organizations.security.giantswarm.io.md" >}})). An organization is defined simply by a custom resource using that CRD, which we'll call an "organization CR" here for brevity.
```
Comment (stone-z) [inline src/content/platform-overview/multi-tenancy/index.md:92]:
```suggestion
Giant Swarm management clusters provide a CRD named `Organization` (long form: `organizations.security.giantswarm.io`, [schema documentation]({{< relref "/use-the-api/management-api/crd/organizations.security.giantswarm.io.md" >}})). An organization is defined simply by a custom resource using that CRD, which we'll call an "organization CR" here for brevity.
```

**Review by stone-z — APPROVED** [review]
Some suggestions but overall LGTM

**Review by pipo02mix — APPROVED** [review]
LGTM when Zach's additions are considered

**Pull request merged**